src/infra/repo/MongoUserRepository.ts

```
@@ -28,17 +57,46 @@ export class MongoUserRepository implements IUserRepository {
+    return mappedUsers;
+  }
+
   async getAllUsersInfo(): Promise<UserInfo[]> {
     return this.findAndMapToUserInfo({});
   }
 
-  async update(userId: string, userData: Partial<User>): Promise<void> {
-    await UserModel.findByIdAndUpdate(userId, { $set: userData }).exec();
+  async update(id: string, update: Partial<User>): Promise<User | null>;
+  async update(userId: string, userData: Partial<User>): Promise<void>;
+  async update(id: string, update: Partial<User>): Promise<User | null | void> {
```

Biome (1.9.4)

[error] void is confusing inside a union type.

Unsafe fix: Use undefined instead.


(lint/suspicious/noConfusingVoidType)